content/techniques/raw-body.md

```diff
@@ -0,0 +1,61 @@
+### Raw body
+
+Nest applications have an option to provide the raw contents of the request body. It is disabled by default.
+
+One of the most common use-case for having access to the raw request body is performing webhook signature verifications. Usually to perform webhook signature validations the unserialized request body is required to calculate an HMAC hash.
+
+#### Use with Express
+
+First enable the option when creating your Nest Express application:
+
+```typescript
+const app = await NestFactory.create(AppModule, {
+  rawBody: true,
+});
+await app.listen(3000);
+```
+
+To access the raw request body in a controller, a convenience interface `RawBodyRequest` is provided to expose a `rawBody` field on the request: use the interface `RawBodyRequest` type:
+
+```typescript
+import { Controller, Post, RawBodyRequest, Req } from '@nestjs/common';
+import { Request } from 'express';
+
+@Controller('cats')
+class CatsController {
+  @Post()
+  create(@Req() req: RawBodyRequest<Request>) {
+    const raw = req.rawBody; // returns a `Buffer`.
+  }
+}
+```
+
+#### Use with Fastify
+
+First enable the option when creating your Nest Fastify application:
+
+```typescript
+const app = await NestFactory.create<NestFastifyApplication>(
+  AppModule,
+  new FastifyAdapter()
+  {
+    rawBody: true,
+  }
+);
+await app.listen(3000);
+```
+
+To access the raw request body in a controller, a convenience interface `RawBodyRequest` is provided to expose a `rawBody` field on the request: use the interface `RawBodyRequest` type:
+
+```typescript
+import { Controller, Post, RawBodyRequest, Req } from '@nestjs/common';
+import { FastifyRequest } from 'fastify';
+
+@Controller('cats')
+class CatsController {
+  @Post()
+  create(@Req() req: RawBodyRequest<FastifyRequest>) {
```

Comment:
as another option, can't we use `@Req('rawBody')` while keeping that parameter type-safe? :thinking:

Reply:
maybe

```ts
@Req('rawBody') rawBody: RawBodyRequest['rawBody']
```

should work?

Reply:
I feel like `RawBodyRequest<FastifyRequest>` looks cleaner than `RawBodyRequest['rawBody']` but that's just my personal preference.